lib/rubocop/cop/lint/redundant_require_statement.rb

```diff
@@ -48,6 +48,8 @@ def on_send(node)
           return unless redundant_require_statement?(node)
 
           add_offense(node) do |corrector|
+            next if node.parent.respond_to?(:modifier_form?) && node.parent.modifier_form?
```

Comment:
If we're confident no-one will do `require "foo" until x` maybe we can be more specific and check if `parent` is an `if_type?`. WDYT? 🙂

Reply:
We probably need a lint about `require... until` :D That's some wild code! :D

Reply:
Needless to say I have never seen it in the wild. I was trying to dream up some use case for it, but can honestly say I failed to even imagine one. 😅

Reply:
Same here! 😆